c++/test/tarray.cpp

```diff
@@ -182,7 +182,7 @@ test_array_compound_array()
         // Check the array dimensions
         for (ii = 0; ii < ndims; ii++)
             if (rdims1[ii] != tdims1[ii]) {
-                TestErrPrintf("Array dimension information doesn't match!, rdims1[%d]=%zd, tdims1[%d]=z%d\n",
+                TestErrPrintf("Array dimension information doesn't match!, rdims1[%d]=%lld, tdims1[%d]=%lld\n",
```

Comment:
I'm not sure about `%lld`.... There's definitely a disagreement in signedness. `"%" PRIuHSIZE` will always work to format an `hsize_t`.

Reply:
Indeed the signedness was and is wrong, will fix.